src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -168,7 +184,20 @@ public void setExpense(Expense target, Expense editedExpense) {
         wallet.setExpense(target, editedExpense);
     }
 
-    // =========== Util Methods ===============================================================================
+    @Override
+    public void deleteTransaction(Transaction transactionToDelete) {
+        if (transactionToDelete instanceof Expense) {
+            deleteExpense((Expense) transactionToDelete);
+        } else if (transactionToDelete instanceof Income) {
+            deleteIncome((Income) transactionToDelete);
+        } else {
+            assert transactionToDelete instanceof Income || transactionToDelete instanceof Expense
```

Comment:
I don't think should use assert here. Should throw instead, but seems like ModelManager is not supposed to throw anything. May have to change the method/do this checking in some other class. I'm not too sure too.

Reply:
If we leaving it as the assert pattern, I think the recommended usage is

if (transactionToDelete instanceof Expense) {
    deleteExpense
} else {
    assert transactionToDelete instanceof Income
    deleteIncome

Perhaps this checking should be moved into the Wallet instead, so that if we do change the impl of the wallet then we would only have to change code in one location.

Reply:
I left the assert there since the quality guide said

> you can use the 'default' branch to detect errors (i.e. if execution reached the default branch, throw an exception). This also applies to the final else of an if-else construct. That is, the final else should mean 'everything else', and not the final option. Do not use else when an if condition can be explicitly specified, unless there is absolutely no other possibility.

Maybe I should follow Joel's implementation instead?